extensions/pip/main.ts

```diff
@@ -0,0 +1,214 @@
+const knownSubcommands = [
+  "install",
+  "download",
+  "uninstall",
+  "freeze",
+  "inspect",
+  "list",
+  "show",
+  "check",
+  "config",
+  "search",
+  "cache",
+  "index",
+  "wheel",
+  "hash",
+  "completion",
+  "debug",
+  "help",
+];
+
+// Ensure the first argument is a known subcommand.
+//
+// This prevents us from skipping the analysis when an argument is passed before
+// the first subcommand (i.e.: `pip --no-color install package`).
+const subcommand = Deno.args[0];
+if (Deno.args.length != 0 && !knownSubcommands.includes(subcommand)) {
+  console.error(
+    `[${
+      red("phylum")
+    }] This extension does not support arguments before the first subcommand. Please open an issue if "${subcommand}" is not an argument.`,
+  );
+  Deno.exit(125);
+}
+
+// Ignore all commands that shouldn't be intercepted.
+if (Deno.args.length == 0 || subcommand != "install") {
+  const cmd = Deno.run({ cmd: ["pip3", ...Deno.args] });
+  const status = await cmd.status();
+  Deno.exit(status.code);
+}
+
+// Analyze new dependencies with phylum before install/update.
+await checkDryRun();
+
+// Perform the package installation.
+const installStatus = PhylumApi.runSandboxed({
+  cmd: "pip3",
+  args: Deno.args,
+  exceptions: {
+    run: ["./", "/bin", "/usr/bin", "/usr/local/bin", "~/.pyenv"],
+    write: [
+      "./",
+      "~/Library/Caches",
+      "~/Library/Python",
+      "~/.cache",
+      "~/.local",
+      "~/.pyenv",
+      "/tmp",
+    ],
+    read: [
+      "~/Library/Caches",
+      "~/Library/Python",
+      "~/.cache",
+      "~/.local",
+      "~/.pyenv",
+      "/tmp",
+      "/etc/passwd",
+      "/etc/apache2/mime.types",
+    ],
+    net: true,
+  },
+});
+Deno.exit(installStatus.code ?? 255);
+
+type JobStatus = {
+  packages: {
+    issues: {
+      severity: string;
+      title: string;
+    }[];
+  }[];
+};
+
+// Logs any identified issues to the screen.
+function logIssues(jobStatus: Record<string, unknown>) {
+  for (const pkg of (jobStatus as JobStatus).packages) {
+    const issues = pkg.issues;
+
+    for (const issue of issues) {
+      let severity = issue.severity.toLowerCase();
+
+      if (["high", "critical"].indexOf(severity) != -1) {
+        severity = red(severity);
+      } else if (severity == "medium") {
+        severity = yellow(severity);
+      } else {
+        severity = green(severity);
+      }
+
+      console.log(`    [${severity}] ${issue.title}`);
+    }
+  }
+}
+
+// Analyze new packages.
+async function checkDryRun() {
+  console.log(`[${green("phylum")}] Finding new dependencies…`);
+
+  const status = PhylumApi.runSandboxed({
+    cmd: "pip3",
+    args: [...Deno.args, "--quiet", "--report", "-", "--dry-run"],
+    exceptions: {
+      run: ["./", "/bin", "/usr/bin", "/usr/local/bin", "~/.pyenv"],
+      write: [
+        "~/Library/Caches",
+        "~/.pyenv",
+        "~/.cache",
+        "~/.local/lib",
+        "/tmp",
+      ],
+      read: [
```

Comment:
The `read` entries here don't match what is in the manifest. Is that intended?

Reply:
Yes. The `--dry-run` call doesn't require the same permissions as the actual install